nvtabular/workflow.py

```diff
@@ -178,10 +190,61 @@ def fit_transform(self, dataset: Dataset) -> Dataset:
         -------
         Dataset
         """
+        cached = DatasetCollection(data=dataset).load_transformed_from_dir(self.work_dir, self)
+        if cached:
+            return cached.data
+
+        LOG.info("Fitting dataset...")
         self.fit(dataset)
-        return self.transform(dataset)
 
-    def save(self, path):
+        LOG.info("Transforming dataset...")
+        transformed = self.transform(dataset)
+
+        if save:
+            DatasetCollection(data=dataset).to_parquet(self.work_dir, overwrite=overwrite, **kwargs)
+
+        return transformed
+
+    def generate_schema(self, dataset, output_path=None):
+        return Schema.calculate_on_dataset(
+            dataset, self.column_group.tags_by_column(), output_path=output_path
+        )
+
+    def fit_transform_collection(
```

Comment:
I wonder if we could make `fit_transform` accept and process a variable number of `Datasets` and bypass the need for a second method here?